### ISSUE TITLE
codemeta:funding

### ISSUE THREAD

**Issue by cboettig** [codemeta/codemeta#137]
We currently define `codemeta:funding` as a text string, e.g.

```
    "funding": "National Science Foundation grant #012345678",
```

Worth noting that schema.org SoftwareSourceCode already has the related notion of `funder`, though it's expected type is an organization (which might be better, though perhaps not well adapted to specifying things like particular NSF grant ids).  Thoughts @danielskatz @mfenner others?

**Comment by sdruskat** [comment]
Could you not have `funder`, and `funder.grantId` to catch both, where possible?

**Comment by danielskatz** [comment]
At some point in the future, we will likely have IDs for funder orgs (ala ORCID), so I think this is a bit uncertain now, but @sdruskat 's idea of separation makes sense to me.

**Comment by mfenner** [comment]
I think that [Open Funder Registry](https://www.crossref.org/services/funder-registry/) is a pretty solid implementation of Funder IDs, using DOIs. So a slight modification could be:

```
"funding": {
  "@type": "Organization",
  "@id": "https://doi.org/10.13039/100000001",
  "name": "National Science Foundation"
}
```
Grant information adds complexity.

**Comment by cboettig** [comment]
The use of Open Funder Registry ids is a nice way to provide `@id`s to the [schema:funder](http://schema.org/funder) (a person or organization).  

 I propose we could just stick with adopting `schema:funder` (the person/organization) into codemeta (since it's already part of `schema:SoftwareSourceCode` which we're extending), and that we also keep `codemeta:funding` (the text string or URL of the particular grant or award).  We 
recommend describing Funder as an Organization using Open Funder Registry, and including a grant URL when applicable under the separate `funding` property.  

Sound reasonable?

**Comment by cboettig** [comment]
Both terms are available in #142 proposal.  (Assuming `funder` will see more use as it is part of schema.org, can take either a Open Funder Registry (or other) ID or a Person/Organization node, but `funding` is provided in the codemeta extension to be used when necessary, e.g. grant numbers).

**Comment by mfenner** [comment]
Several of us are working on extending schema.org for better funding support, in particular grant IDs. We hope to have something stable in June and will then propose to the schema.org community.

Basic idea is currently to turn the Schema.org "Award" into a subclass, which can then be linked to a "CreativeWork" via for example "funding", and has the inherited properties name, identifier, etc. Would be good to align.
